src/main/java/com/uber/cadence/migration/MigrationInterceptor.java

```diff
@@ -0,0 +1,142 @@
+            Workflow.sideEffect(
+                MigrationDecision.class, () -> shouldMigrate(workflowDefinition, input));
+        if (decision.shouldMigrate) {
+          StartWorkflowExecutionRequest request =
+              new StartWorkflowExecutionRequest()
+                  .setDomain(workflowInfo.getDomain())
+                  .setWorkflowId(workflowInfo.getWorkflowId())
+                  .setTaskList(
+                      new TaskList()
+                          .setName(
+                              input
+                                  .getWorkflowExecutionStartedEventAttributes()
+                                  .taskList
+                                  .getName()))
+                  .setInput(input.getInput())
+                  .setWorkflowType(new WorkflowType().setName(input.getWorkflowType().getName()))
+                  .setWorkflowIdReusePolicy(WorkflowIdReusePolicy.TerminateIfRunning)
+                  .setRetryPolicy(
+                      input.getWorkflowExecutionStartedEventAttributes().getRetryPolicy())
+                  .setRequestId(UUID.randomUUID().toString())
+                  .setIdentity(input.getWorkflowExecutionStartedEventAttributes().getIdentity())
+                  .setMemo(input.getWorkflowExecutionStartedEventAttributes().getMemo())
+                  .setCronSchedule(
+                      input.getWorkflowExecutionStartedEventAttributes().getCronSchedule())
+                  .setDelayStartSeconds(
+                      input
+                          .getWorkflowExecutionStartedEventAttributes()
+                          .getFirstDecisionTaskBackoffSeconds())
+                  .setHeader(input.getWorkflowExecutionStartedEventAttributes().getHeader())
+                  .setSearchAttributes(
+                      input.getWorkflowExecutionStartedEventAttributes().getSearchAttributes())
+                  .setExecutionStartToCloseTimeoutSeconds(
+                      input
+                          .getWorkflowExecutionStartedEventAttributes()
+                          .getExecutionStartToCloseTimeoutSeconds())
+                  .setTaskStartToCloseTimeoutSeconds(
+                      input
+                          .getWorkflowExecutionStartedEventAttributes()
+                          .getTaskStartToCloseTimeoutSeconds());
+
+          try {
+            MigrationActivities.StartNewWorkflowExecutionResponse response =
+                activities.startWorkflowInNewDomain(
+                    new MigrationActivities.StartNewWorkflowRequest());
+            // TODO: add metrics and logging
+            throw new CancellationException(
+                "cancel due to migration:" + response.response.toString());
+          } catch (ActivityException e) {
+            // fallback if start workflow in new domain failed
+            next.executeWorkflow(workflowDefinition, input);
+          }
+        }
+    }
+    return next.executeWorkflow(workflowDefinition, input);
```

Comment:
delete this line, let's conclude return in every switch branch
```suggestion
```

Reply:
yeah, tried that. but even if we have default that returns the next, it shows missing return statement